File: src/main/java/io/github/divyeshmangla/timetable/model/ClassType.java
package io.github.divyeshmangla.timetable.model;

public enum ClassType {
    LECTURE('L'),
    TUTORIAL('T'),
    PRACTICAL('P'),
    ;

    private final char suffix;

    ClassType(char suffix) {
        this.suffix = suffix;
    }

    public char getSuffix() {
        return suffix;
    }

    public static ClassType fromSuffix(char suffix) {
        for (ClassType type : values()) {
            if (type.suffix == suffix) {
                return type;
            }
        }
        return null;
    }
}

File: internal/model/class_type.go
package model

type ClassType int

const (
	LECTURE ClassType = iota
	TUTORIAL
	PRACTICAL
)

var classTypeSuffixes = map[ClassType]rune{
	LECTURE:   'L',
	TUTORIAL:  'T',
	PRACTICAL: 'P',
}

var suffixToClassType = map[rune]ClassType{
	'L': LECTURE,
	'T': TUTORIAL,
	'P': PRACTICAL,
}

func (ct ClassType) Suffix() rune {
	return classTypeSuffixes[ct]
}

func FromSuffix(suffix rune) *ClassType {
	if ct, ok := suffixToClassType[suffix]; ok {
		return &ct
	}

	return nil
}

func (ct ClassType) String() string {
	switch ct {
	case LECTURE:
		return "LECTURE"
	case TUTORIAL:
		return "TUTORIAL"
	case PRACTICAL:
		return "PRACTICAL"
	default:
		return "UNKNOWN"
	}
}
